src/wp-admin/user-edit.php

```diff
@@ -241,213 +255,225 @@
 <h2 class="user-profile-personal-options"><?php _e( 'Personal Options' ); ?></h2>
 
 <table class="form-table">
-<?php if ( ! ( IS_PROFILE_PAGE && ! $user_can_edit ) ) : ?>
+		<?php if ( ! ( IS_PROFILE_PAGE && ! $user_can_edit ) ) : ?>
 	<tr class="user-rich-editing-wrap">
 		<th scope="row"><?php _e( 'Visual Editor' ); ?></th>
-		<td><label for="rich_editing"><input name="rich_editing" type="checkbox" id="rich_editing" value="false" <?php if ( ! empty( $profileuser->rich_editing ) ) checked( 'false', $profileuser->rich_editing ); ?> /> <?php _e( 'Disable the visual editor when writing' ); ?></label></td>
+		<td><label for="rich_editing"><input name="rich_editing" type="checkbox" id="rich_editing" value="false" <?php checked( 'false', $profileuser->rich_editing ); ?> /> <?php _e( 'Disable the visual editor when writing' ); ?></label></td>
 	</tr>
 <?php endif; ?>
-<?php
-$show_syntax_highlighting_preference = (
-	// For Custom HTML widget and Additional CSS in Customizer.
-	user_can( $profileuser, 'edit_theme_options' )
-	||
-	// Edit plugins.
-	user_can( $profileuser, 'edit_plugins' )
-	||
-	// Edit themes.
-	user_can( $profileuser, 'edit_themes' )
-);
-?>
-<?php if ( $show_syntax_highlighting_preference ) : ?>
+		<?php
+		$show_syntax_highlighting_preference = (
+		// For Custom HTML widget and Additional CSS in Customizer.
+		user_can( $profileuser, 'edit_theme_options' )
+		||
+		// Edit plugins.
+		user_can( $profileuser, 'edit_plugins' )
+		||
+		// Edit themes.
+		user_can( $profileuser, 'edit_themes' )
+		);
+		?>
+		<?php if ( $show_syntax_highlighting_preference ) : ?>
 	<tr class="user-syntax-highlighting-wrap">
 		<th scope="row"><?php _e( 'Syntax Highlighting' ); ?></th>
 		<td>
-			<label for="syntax_highlighting"><input name="syntax_highlighting" type="checkbox" id="syntax_highlighting" value="false" <?php if ( ! empty( $profileuser->syntax_highlighting ) ) checked( 'false', $profileuser->syntax_highlighting ); ?> /> <?php _e( 'Disable syntax highlighting when editing code' ); ?></label>
+			<label for="syntax_highlighting"><input name="syntax_highlighting" type="checkbox" id="syntax_highlighting" value="false" <?php checked( 'false', $profileuser->syntax_highlighting ); ?> /> <?php _e( 'Disable syntax highlighting when editing code' ); ?></label>
```

Comment:
The check for `! empty()` was removed?

Reply:
It does seem this was intentional in the upstream changeset that was backported.
https://core.trac.wordpress.org/changeset/42217#file12

Reply:
The check must have been deemed unnecessary since the value is always set to false in HTML.

![Screenshot 2022-07-23 at 17 39 19](https://user-images.githubusercontent.com/7713923/180609995-14b9cfc8-c427-4a6e-9bf9-0b0b4239a64a.png)

Reply:
A value of "false" is not the same as `false` detected by `empty()`.

Reply:
The data types are definitely different. However, the if check doesn't check for the value being set and data type being same; It goes to find if the value has been set and that will always be set as "true".

Reply:
I see that since it's checking the value in the database, not the value from the form, and the database contains a string instead of a boolean, then the `empty` check does nothing.

Reply:
Resolving this since the previous behavior of the code is still intact.